### PR TITLE
WPF solution generator

### PR DESCRIPTION
**Pull request by GantMan**
This PR provides a way to run WPF as a separate solution.

A user can generate a WPF project, and run it from the command line.

Once this is in place, we have a functioning WPF and UWP structure.  This is the cornerstone for being able to restructure those two parts for consolidating javascript in `react-native-windows-core`

Please review and let me know any examples/videos/gifs you might want.

### CODE REVIEW

**Comment by rozele** [comment]
> 

I don't remember if we talked about this or not.  Are we going to create a second rnpm-plugin-wpf command for this, or were we just going to use rnpm-plugin-windows?

---
Refers to: local-cli/rnpm/windows/index.js:26 in d1b60b4. [](commit_id = d1b60b4c1fa886410751fc72fcbe4dc9631542ad, deletion_comment = False)

File: local-cli/rnpm/windows/src/wpf.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+module.exports = function windows(config, args, options) {
+  const name = args[0] ? args[0] : Common.getReactNativeAppName();
+  const ns = options.namespace ? options.namespace : name;
+  const version = options.windowsVersion ? options.windowsVersion : Common.getReactNativeVersion();
+
+  return Common.getInstallPackage(version)
+    .then(rnwPackage => {
+      console.log(`Installing ${rnwPackage}...`);
+      // execSync(`npm install --save ${rnwPackage}`);
+      console.log(chalk.green(`${rnwPackage} successfully installed.`));
```
Comment (rozele) [inline local-cli/rnpm/windows/src/wpf.js]: Comment required?

**Comment by rozele** [comment]
> 

I see, you're just sharing code.

---
In reply to: [272306140](https://github.com/ReactWindows/react-native-windows/pull/954#issuecomment-272306140) [](ancestors = 272306140)

---
Refers to: local-cli/rnpm/windows/index.js:26 in d1b60b4. [](commit_id = d1b60b4c1fa886410751fc72fcbe4dc9631542ad, deletion_comment = False)

File: local-cli/rnpm/windows/src/common.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  return path.resolve(
+    process.cwd(),
+    'node_modules',
+    'react-native-windows',
+    'local-cli',
+    'generate-windows.js'
+  );
+};
+
+const REACT_NATIVE_WPF_GENERATE_PATH = function() {
```
Comment (rozele) [inline local-cli/rnpm/windows/src/common.js]: >REACT_NATIVE_WPF_GENERATE_PATH [](start = 6, length = 30)

Can we put WPF specific stuff in the wpf.js and UWP specific stuff back in windows.js?

File: local-cli/runWpf/runWpf.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    description: 'Deploys the app to an emulator',
+  }, {
+    command: '--device',
+    description: 'Deploys the app to a connected device',
+  }, {
+    command: '--target [string]',
+    description: 'Deploys the app to the specified GUID for a device.',
+  }, {
+    command: '--proxy',
+    description: 'Deploys the app in remote debugging mode.',
```
Comment (rozele) [inline local-cli/runWpf/runWpf.js]: Clean up a bunch of these command parameters that no longer make sense.

File: package.json [hunk 4]
```diff
@@ -19,6 +19,7 @@
     "Libraries",
     "ReactWindows/ReactNative",
     "ReactWindows/ReactNative.Shared",
+     "ReactWindows/ReactNative.Net46",
     "ReactWindows/ChakraBridge",
```
Comment (rozele) [inline package.json]: This isn't supposed to go in the package.json yet.  Or is this just a temporary step to unblock until you create the build process for separate packages?

**Comment by rozele** [comment]
:clock1:

**Comment by GantMan** [comment]
**Regarding:**
> I don't remember if we talked about this or not. Are we going to create a second rnpm-plugin-wpf command for this, or were we just going to use rnpm-plugin-windows?

Short answer: Simplest for end users seems to be both inside `rnpm-plugin-windows` module.

Long Answer:  I have written both.  I have a plugin called `rnpm-plugin-wpf` (which I added to you as publisher on npmjs), and basically is as simple as the original.  I used it during testing. The resulting "getting started" directions would be more complicated for no real benefit, so I've got the code added to `rnpm-plugin-windows` in this PR.  

**Regarding:** 
> This isn't supposed to go in the package.json yet. Or is this just a temporary step to unblock until you create the build process for separate packages?

Yup, temporary step to unblock.  We can use this just fine until the separate packages is ready to take over.

**Everything Else:**
Feedback integrated, PR updated.

**Comment by rozele** [comment]
:shipit:

**Pull request merged**